src/NHibernate/Collection/Generic/SetHelpers/SetSnapShot.cs

```diff
@@ -114,12 +114,22 @@ IEnumerator IEnumerable.GetEnumerator()
 
 		void ICollection.CopyTo(Array array, int index)
```

Comment:
You've forgot about overload for .NET Core https://github.com/nhibernate/nhibernate-core/blob/2ee35131208be016cc038fafc62b6c55d69d59dc/src/NHibernate/Collection/Generic/SetHelpers/SetSnapShot.cs#L164-L172